lib/api/route/messages.dart

```diff
@@ -158,6 +158,19 @@ class GetMessagesResult {
 // https://zulip.com/api/send-message#parameter-topic
 const int kMaxTopicLength = 60;
 
+/// The canonical resolved-topic prefix.
+// Adapted from the shared code:
+//   https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts
+const String kResolvedTopicPrefix = '✔ ';
+
+String stripResolvePrefixIfPresent(String topic) =>
+  topic.startsWith(kResolvedTopicPrefix)
+    ? topic.substring(kResolvedTopicPrefix.length)
+    : topic;
```

Comment:
How does this end up getting used later in your branch?

Its behavior differs from `unresolve_name` in the shared code:
https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts#L22-L29

which I'm guessing (haven't checked) is what gets applied when the user unresolves a topic in web or the legacy mobile app.